src/overlays/actors/ovl_Bg_Po_Syokudai/z_bg_po_syokudai.c

```diff
@@ -21,11 +46,128 @@ const ActorInit Bg_Po_Syokudai_InitVars = {
+    this->collider.dim.pos.y = (s16)this->actor.posRot.pos.y;
+    this->collider.dim.pos.z = (s16)this->actor.posRot.pos.z;
+
+    if (!this->unk_14C && Flags_GetSwitch(globalCtx, 0x1F) && Flags_GetSwitch(globalCtx, 0x1E) &&
+        Flags_GetSwitch(globalCtx, 0x1D) && !Flags_GetSwitch(globalCtx, this->actor.params)) {
+
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_PO_SISTERS, 119.0f, 225.0f, -1566.0f, 0, 0, 0,
+                    this->actor.params);
+        globalCtx->envCtx.unk_BF = 0x4U;
+
+    } else if (!Flags_GetSwitch(globalCtx, 0x1C) && !Flags_GetSwitch(globalCtx, 0x1B)) {
+
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_PO_SISTERS, this->actor.posRot.pos.x,
+                    this->actor.posRot.pos.y + 52.0f, this->actor.posRot.pos.z, 0, 0, 0,
+                    (((u8)this->unk_14C) << 8) + this->actor.params + 0x1000);
+
+    } else if (!Flags_GetSwitch(globalCtx, this->actor.params)) {
+        if (globalCtx->envCtx.unk_BF == 0xFF) {
+            globalCtx->envCtx.unk_BF = 0x4U;
+        }
+    }
+
+    this->unk_14E = (s16)(Math_Rand_ZeroOne() * 20.0f);
+}
+
+void BgPoSyokudai_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgPoSyokudai* this = THIS;
+
+    LightContext_RemoveLight(globalCtx, &globalCtx->lightCtx, this->lightNode);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+
+    if (globalCtx->envCtx.unk_BF != 0xFF) {
+        globalCtx->envCtx.unk_BF = (u8)0xFFU;
+    }
+}
+
+void BgPoSyokudai_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgPoSyokudai* this = THIS;
+
+    void* pad;
+
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    if (Flags_GetSwitch(globalCtx, this->actor.params)) {
+        func_8002F974(&this->actor, NA_SE_EV_TORCH - SFX_FLAG);
+    }
+    this->unk_14E += 1;
+}
+
+void BgPoSyokudai_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgPoSyokudai* this = THIS;
+
+    f32 lightBrightness;
+    u8 red;
+    u8 green;
+    u8 blue;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_po_syokudai.c", 315);
+
+    func_80093D18(globalCtx->state.gfxCtx);
+    gSPMatrix(oGfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_po_syokudai.c", 319),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(oGfxCtx->polyOpa.p++, D_060003A0);
+
+    if (Flags_GetSwitch(globalCtx, this->actor.params)) {
+        Color_RGBA8* primColor = &primColors[this->unk_14C];
+        Color_RGBA8* envColor = &envColors[this->unk_14C];
+
+        lightBrightness = (0.3f * Math_Rand_ZeroOne()) + 0.7f;
+
+        red = ((u32)(f32)(primColor->r * lightBrightness)) & 0xFF;
+        green = ((u32)(f32)(primColor->g * lightBrightness)) & 0xFF;
+        blue = ((u32)((f32)primColor->b * lightBrightness)) & 0xFF;
+
+        Lights_PointSetColorAndRadius(&this->lightInfo, red, green, blue, 0xC8);
```

Comment:
`Lights_PointSetColorAndRadius`'s last argument `0xC8` is a radius so it's better as decimal